### PR TITLE
[FIX JENKINS-42008] plugin-pom should embed the sha and branch into the plugins MANIFEST

### PR DESCRIPTION
**Pull request by batmat**
This change adds 2 lines like the following to the `MANIFEST.MF` of `hpi`s:

```
Branch: master
CommitId: 046af1b646832fdb3f4945ad71b8e30310184b3c
```

@jenkinsci/code-reviewers and @andresrc @stephenc @jglick @jtnord at least

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -472,6 +472,23 @@
         </executions>
       </plugin>
       <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>buildnumber-maven-plugin</artifactId>
```
Comment (KostyaSha) [inline pom.xml]: Maybe you want maven-git-commit-id-plugin ?

Comment (batmat) [inline pom.xml]: @KostyaSha I prefer using things coming from MojoHaus when possible for many reasons :-). Also, I really want to put those into the MANIFEST, is this feasible with this one? After a 30 seconds look, I'm not sure it is.

**Comment by batmat** [comment]
@reviewbybees for more eyes.

**Comment by Vlatombe** [comment]
🐝

**Comment by jglick** [comment]
The commit hash seems superfluous given [this code](https://github.com/jenkinsci/maven-hpi-plugin/blob/b927bea438884c74b353b27655aeaedfd99be4e1/src/main/java/org/jenkinsci/maven/plugins/hpi/AbstractHpiMojo.java#L878-L946), and the branch seems superfluous given the commit hash—right? What problem are you trying to solve here?

**Comment by oleg-nenashev** [comment]
@batmat  I suppose the comment from @jglick should be answered before merging

**Comment by oleg-nenashev** [comment]
@batmat gentle ping

**Comment by batmat** [comment]
Sorry, lagging those days. I think what Jesse pointed out most probably makes this unnecessary. Hence I'm closing this, and will reopen if/when I find time to look into specifics here.
Thanks for the review everyone.

**Pull request closed without merging**